kernel/src/mm/pagetable.rs

```diff
@@ -361,6 +362,11 @@ impl PTEntry {
         let addr = PhysAddr::from(self.0.bits() & 0x000f_ffff_ffff_f000);
         strip_confidentiality_bits(addr)
     }
+
+    /// Read a page table entry from the specified virtual address.
+    pub fn read_pte(vaddr: VirtAddr) -> Self {
+        unsafe { *vaddr.as_ptr::<Self>() }
+    }
```

Comment:
```suggestion
    pub unsafe fn read_pte(vaddr: VirtAddr) -> Self {
        *vaddr.as_ptr::<Self>()
    }
```

That was the more important part of my [suggestion](https://github.com/coconut-svsm/svsm/pull/461#discussion_r1771258416) regarding this function :D

Reply:
Sorry I missed that before - completely agree.